services/app-web/src/components/Form/ErrorSummary/ErrorSummary.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames'
+import classnames from 'classnames'
+import React, { useEffect, useRef } from 'react'
+import styles from './ErrorSummary.module.scss'
+
+type ErrorSummaryMessageProps = {
+    errorKey: string,
+    message: string
+}
+const ErrorSummaryMessage = ({
+    errorKey, message
+}: ErrorSummaryMessageProps): React.ReactElement => {
+    return (
+        <button
+            type="button"
+            className={classnames("usa-error-message", "usa-alert__text", styles.message)}
+            onClick={() => {
+                const fieldElement: HTMLElement | null = document.querySelector(
+                    `[name="${errorKey}"]`
+                );
+            
+                if (fieldElement) {
+                    fieldElement.focus();
+                }
+            }}
+        >
+        {message}
+      </button> 
+    ) 
+}
+
+function summaryHeading(numberOfErrors: number) {
+    if (numberOfErrors === 1) {
+        return "There is 1 error on this page";
+    } else {
+        return `There are ${numberOfErrors} errors on this page`;
+    }
+}
+
+function flattenMessage(message: string | string[]) {
+    if (typeof message === "string") {
+        return message;
+    }
+
+    return message.join(" and ");
+}
+
+export type ErrorSummaryProps = {
+    errors: { [field: string]: string | string[] }
+}
+
+/**
+ * This component renders a list of validation errors.
+ * 
+ * Each message will render a button that will focus the relevant field
+ * when it is clicked.
+ *
+ * This component relies on the errors object returned by Formik, so it should generally
+ * rendered inside of a Formik form context.
+ */
+export const ErrorSummary = ({
+    errors
+}: ErrorSummaryProps): React.ReactElement | null => {
+    const numberOfErrors = Object.keys(errors).length;
+
+    const headingRef = useRef<HTMLHeadingElement>(null);
+
+    useEffect(() => {
+        if (numberOfErrors === 0) {
+            const { current } = headingRef;
+            if (current) {
+                current.focus();
```

Comment:
This will focus the heading when there are errors. I still need to verify this is working properly in screen readers.

Reply:
The  assertion confused me- based on what its doing seems like it would be number of !==0 to run the effect

Reply:
Yeah, that is absolutely a bug. It should be `numberOfErrors > 0`!

Reply:
Since this effect is being run every time the number of errors changes, is there a risk here that we're going to focus the error summary as we're just filling in form fields, after having clicked "continue" once? I'm waiting for a deploy to play with it live but I'm curious about this.